internal/gitinterface/keys.go

```diff
@@ -257,3 +268,66 @@ func verifyGitsignSignature(ctx context.Context, key *tuf.Key, data, signature [
 
 	return nil
 }
+
+// verifySSHKeySignature verifies Git signatures issued by SSH keys.
+func verifySSHKeySignature(key *tuf.Key, data, signature []byte) error {
+	verifier, err := signerverifier.NewSignerVerifierFromTUFKey(key)
+	if err != nil {
+		return errors.Join(ErrVerifyingSSHSignature, err)
+	}
+
+	publicKey, err := ssh.NewPublicKey(verifier.Public())
+	if err != nil {
+		return errors.Join(ErrVerifyingSSHSignature, err)
+	}
+
+	sshSignature, err := decodeSSHSignature(signature)
+	if err != nil {
+		return errors.Join(ErrVerifyingSSHSignature, err)
+	}
+
+	if err := publicKey.Verify(data, sshSignature); err == nil {
+		return nil
+	}
+
+	return ErrIncorrectVerificationKey
```

Comment:
```suggestion
	if err := publicKey.Verify(data, sshSignature); err != nil {
		return errors.Join(ErrIncorrectVerificationKey, err)
	}
	return nil
```

we probably want to join the errors here too, this way we don't drop any debug information about why the verify may have failed.

Reply:
Done! Took the liberty of updating this in verifyGitsignSignature as well.